### DISCUSSION TITLE
WG_PORT Wasn't what I thought it was

### DISCUSSION THREAD

**Discussion by Sazerak** [wg-easy/wg-easy#1309] [General]
Many moons ago when I first setup wg-easy, I used WG_PORT to identify the OUTSIDE port for my docker setup, so my config looked like this:
```
<more config>
- WG_PORT=12345
<more config>
ports:
      - 12345:51820/udp
      - 51821:51821/tcp
<more config>
```
and this worked perfectly for the longest time. I don't remember why I set it up this way (probably watched a YouTube video or something), but my VPN recently stopped working and I couldn't figure out why. Fast forward to today and I learn that WG_PORT is supposed to identify the INTERNAL port, so I changed the variable and everything works well now.

BUT, this got me wondering: why did my VPN work before? I could only come up with 2 theories. Either the purpose of WG_PORT was changed at some point (I found that unlikely), or WG_PORT never worked until recently.

I figured the last explanation was more likely, so I did some digging and found [this commit](https://github.com/wg-easy/wg-easy/commit/2eec97ff251c1d6e8f2df0bb00d1038cb82a320b) that I think supports my theory. As far as I can tell, Wireguard was always hardcoded to 51820 until the change was made to use WG_PORT only a few months ago. I probably updated my container between then and now, and that's when my problem started.

From the above commit, I was able to find some related discussions which seem to have originally identified the issue and started the chain of events to fix it, so it seems that my hunch was right. It was just sheer back luck that I was misconfigured but that the issue had stayed dormant for so long. Because of this, I never expected that variable to be the cause of my issue.

So that's that, problem solved! I figured I would make this discussion to hopefully help anyone in a similar situation as me who was misconfigured from the start but never realized it since the misconfigured feature never actually worked. Because of this unique situation, it took me a while to track down the issue so if this discussion helps anyone else, it was worth it.

Cheers!

**Comment by offizium-berndstorath** [comment]
Thank you for sharing. This resulted in numerous issues where users complained that their vpn isn't working anymore.
This is unfortunately the result of not properly documentation breaking changes. Its nowhere stated that if you upgrade from version x to y feature z breaks. This will be fixed in the future
